core/src/main/java/com/scalar/db/common/StateManagedDistributedTransactionManager.java

```diff
@@ -21,34 +23,79 @@
 import com.scalar.db.exception.transaction.UnknownTransactionStatusException;
 import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.CopyOnWriteArrayList;
 
-public abstract class TransactionDecorationDistributedTransactionManager
-    extends AbstractDistributedTransactionManager
-    implements DistributedTransactionDecoratorAddable {
+public class StateManagedDistributedTransactionManager
```

Comment:
Ditto. Refactored this from using inheritance to using composition (decorator pattern).